website/static/js/osfHelpers.js

```diff
@@ -857,6 +857,78 @@ function onScrollToBottom(element, callback) {
     });
 }
 
+/**
+ *  The following jquery functions are to fix the bootstrap affix issue
+ *  where having a top offset and a bottom offset can cause conflicts when the page is refreshed.
+ *  In this cause it caused the user settings side navigation bar to stick to the bottom of the page
+ *  when refreshed. The first function checks if the page has been shrunk to the mobile settings or not.
+ *  The second function affixes the element to the top of the screen window and the top of the page footer.
+ *  @param {String} panelID:
+ *  @param {Integer} panel_offset:
+ */
+var stickIt = function stickIt(panelID, panel_offset){
+    this.elementID = panelID;
+    this.offsetElement = panel_offset;
+
+    var lastWidth;
+    $(window).load(function(){
+        if($(window).width() > 769){
+            lastWidth = $(window).width();
+            sticky_element(panelID, panel_offset);
+        }
+    });
+
+    $(window).resize(function(){
+        if($(window).width()!=lastWidth){
+            if(lastWidth > 769 && $(window).width() <= 769){
+                $(window).scroll(function () {
+                    $(elementID).css('top', '');
+                });
+            }
+            lastWidth = $(window).width();
+        }
+
+    });
+
+    $(window).resize(function(){
+        if($(window).width()!=lastWidth){
+            if(lastWidth <= 769 && $(window).width() > 769){
+                lastWidth = $(window).width();
+                sticky_element(panelID, panel_offset);
+            }
+        }
+
+    });
+
+
+    function sticky_element(panelID, panel_offset){
+        var stickyE   = panelID,
+            bottomE   = "#pagefooter";
+        if($( stickyE ).length){
+            $( stickyE ).each(function(){
+                var fromTop = $( this ).offset().top - panel_offset, // number of pixels to the top of element
+                // offset from the top of the screen
+                fromBottom = $( document ).height()-($( this ).offset().top + $( this ).outerHeight()),
+                //number of pixels from the top of the bottom element, adding height to take into account padding/borders
+                stopOn = $( document ).height()-( $( bottomE ).offset().top)+($( this ).outerHeight() - $( this ).height());
+                if( (fromBottom-stopOn) > 200){
```

Comment:
Another magic number. Best to make a constant or a function parameter as mentioned above.